R/findNLDI.R

```diff
@@ -344,67 +354,49 @@ findNLDI <- function(comid = NULL,
   }
 
   # Set empty lists to store origin, navigation, features, and basin requests ...
-  start <- navigate <- features <- basin <- list()
-
-  # Set origin url
-  start[["start"]] <- start_url
+  navigate <- features  <- list()
 
   # Build navigation URLs
   for (i in seq_along(nav)) {
       navigate[[nav[i]]] = paste0(start_url, "navigation/", nav[i])
   }
 
-  # Build basin URL
-  if (getBasin) {
-    basin[["basin"]] = paste0(start_url, "basin")
-  }
-
   # Build find URLs
   if (length(find) > 0) {
-    features = lapply(navigate, paste0, paste0("/", find))
+    features = lapply(navigate, paste0, paste0("/", find), paste0("?distance=", distance_km))
   }
 
-  # Add distance constraints to features
-  features = lapply(features, paste0, paste0("?distance=", distance_km))
 
-  # Add distance constraints to navigations flowpaths
-  navigate = lapply(navigate, paste0, paste0("/flowlines?distance=", distance_km))
+  names  <- unlist(lapply(nav, paste0, paste0("_", find)))
 
-  # combine, unlist, relist
-  ll = as.list(unlist(c(start, basin, navigate, features)))
+  search <- data.frame(
+    url = unlist(c(start_url,
+                   if (getBasin) { paste0(start_url, "basin") },
+                   features)),
 
-  # define the type of each URL (feature, basin, or nav)
-  # This is needed as the attribute names for each varies
-  types = c("feature",
-            if (getBasin) { 'basin' },
-            rep("nav", length(nav)),
-            rep("feature", length(find) * length(nav)))
+    type = c("feature",
```

Comment:
Did you mean to change this to "type" from "types" ?

Reply:
I had to re-write that chunk, I guess I just made them all singular this time haha 🤷‍♂️ . Its consistent with line 338.